pandas/core/series.py

```diff
@@ -3234,40 +3238,22 @@ def _try_cast(arr, take_fast_path):
     else:
         subarr = _try_cast(data, False)
 
-    # scalar like, GH
-    if getattr(subarr, 'ndim', 0) == 0:
-        if isinstance(data, list):  # pragma: no cover
-            subarr = np.array(data, dtype=object)
-        elif index is not None:
+    if subarr.ndim == 0 or is_scalar(data):
+        if subarr.ndim == 1:
+            # a scalar upcasted to 1-dimensional by maybe_cast_to_datetime()
+            value = subarr[0]
```

Comment:
how is this branch ever hit?

Reply:
``pd.Series(pd.Timestamp('2015-10-10'), index=range(3))``